testsuite/MDAnalysisTests/lib/test_util.py

```diff
@@ -245,138 +230,144 @@ class TestMakeWhole(TestCase):
     |           |
     +-----------+
     """
-    def setUp(self):
-        self.u = mda.Universe(Make_Whole)
-        self.ag = self.u.residues[0].atoms
 
-    def tearDown(self):
-        del self.u
+    @staticmethod
+    @pytest.fixture()
+    def universe():
+        return mda.Universe(Make_Whole)
+
+    @staticmethod
+    @pytest.fixture()
+    def ag(universe):
+        return universe.residues[0].atoms
 
-    def test_no_bonds(self):
+    def test_no_bonds(self, ag):
         # NoData caused by no bonds
-        assert_raises(NoDataError, mdamath.make_whole, self.ag)
+        assert_raises(NoDataError, mdamath.make_whole, ag)
 
-    def test_not_orthogonal(self):
+    def test_not_orthogonal(self, universe, ag):
         # Not an orthogonal unit cell
-        self._load_bonds()
+        self._load_bonds(universe)
 
-        self.u.dimensions = [10., 10., 10., 80., 80., 80]
-        assert_raises(ValueError, mdamath.make_whole, self.ag)
+        universe.dimensions = [10., 10., 10., 80., 80., 80]
+        assert_raises(ValueError, mdamath.make_whole, ag)
 
-    def test_zero_box_size(self):
-        self._load_bonds()
+    def test_zero_box_size(self, universe, ag):
+        self._load_bonds(universe)
 
-        self.u.dimensions = [0., 0., 0., 90., 90., 90.]
-        assert_raises(ValueError, mdamath.make_whole, self.ag)
+        universe.dimensions = [0., 0., 0., 90., 90., 90.]
+        with pytest.raises(ValueError):
+            mdamath.make_whole(ag)
 
-    def test_too_small_box_size(self):
-        self._load_bonds()
+    def test_too_small_box_size(self, universe, ag):
+        self._load_bonds(universe)
 
         # Set the z dimensions to 0.5, which is small compared to the
         # bonds (1-2)
-        self.u.dimensions = [100.0, 100.0, 0.5, 90., 90., 90.]
-        assert_raises(ValueError, mdamath.make_whole, self.ag)
+        universe.dimensions = [100.0, 100.0, 0.5, 90., 90., 90.]
+        with pytest.raises(ValueError):
+            mdamath.make_whole(ag)
 
-    def _load_bonds(self):
+    def _load_bonds(self, universe):
```

Comment:
Could we make a fixture called `universe_with_bonds` which is just the universe fixture and this function called?  Ie put the body of this method into a fixture and use that instead?